service/src/main/kotlin/fi/espoo/evaka/dvv/DvvModificationsService.kt

```diff
@@ -95,14 +104,46 @@ class DvvModificationsService(
         }
     }
 
-    private fun handleDeath(db: Database, ssn: String, deathDvvInfoGroup: DeathDvvInfoGroup) = db.transaction { tx ->
-        tx.getPersonBySSN(ssn)?.let {
-            val dateOfDeath = deathDvvInfoGroup.kuolinpv?.asLocalDate() ?: LocalDate.now()
-            logger.info("Dvv modification for ${it.id}: marking dead since $dateOfDeath")
-            tx.updatePersonFromVtj(it.copy(dateOfDeath = dateOfDeath))
+    private fun handleDeath(db: Database, ssn: String, deathDvvInfoGroup: DeathDvvInfoGroup) {
+        if (deathDvvInfoGroup.kuollut != true || deathDvvInfoGroup.kuolinpv == null) return
+
+        db.transaction { tx ->
+            tx.getPersonBySSN(ssn)?.let { person ->
+                val dateOfDeath = deathDvvInfoGroup.kuolinpv.asLocalDate()
+                logger.info("Dvv modification for ${person.id}: marking dead since $dateOfDeath")
+                tx.updatePersonFromVtj(person.copy(dateOfDeath = dateOfDeath))
+
+                endFamilyRelations(tx, person.id, dateOfDeath)
+                asyncJobRunner.plan(tx, listOf(GenerateFinanceDecisions.forAdult(person.id, DateRange(dateOfDeath, null))))
+                asyncJobRunner.plan(tx, listOf(GenerateFinanceDecisions.forChild(person.id, DateRange(dateOfDeath, null))))
+            }
         }
     }
 
+    private fun endFamilyRelations(tx: Database.Transaction, personId: UUID, dateOfDeath: LocalDate) {
+        tx
```

Comment:
What about the guardian table?

Reply:
Isn't mutpa supposed to handle that? If the guardianship ends, mutpa should send an event about that. At least that is my assumption. My opinion is that since it's not something our users have control over anyway, I don't think it should be touched here.